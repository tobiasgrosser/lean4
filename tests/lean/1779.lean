section
variable [Coe Int R] [Neg R] (a : R) (n : Int)
#check a = -n
#check a = (-n : R)
#check -n = a
#check (-n : R) = a
end

section
variable [Coe Int R] [Add R] (a : R) (n : Int)
#check a = n + n
#check a = (n + n : R)
end

section
variable [Coe Int R] [OfNat R 0] [Sub R] (a : R) (n : Int)
#check a = 0 - n
#check a = (0 - n : R)
end

section
variable [Coe Int R] [Add R] [Neg R] (a : R) (n : Int)
#check a + -n
#check a + (-n : R)
#check -n + a
#check (-n : R) + a
end
