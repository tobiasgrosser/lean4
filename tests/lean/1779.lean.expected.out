a = -Coe.coe n : Prop
a = -Coe.coe n : Prop
-Coe.coe n = a : Prop
-Coe.coe n = a : Prop
a = Coe.coe n + Coe.coe n : Prop
a = Coe.coe n + Coe.coe n : Prop
a = 0 - Coe.coe n : Prop
a = 0 - Coe.coe n : Prop
a + -Coe.coe n : R
a + -Coe.coe n : R
-Coe.coe n + a : R
-Coe.coe n + a : R
